webui/react/src/pages/ExperimentDetails/CodeViewer/CodeViewer.tsx

```diff
@@ -0,0 +1,269 @@
+        workspace,
+        project,
+        ...restConfig
+      } = configRaw;
+      return { environment: restEnvironment, ...restConfig };
+    }
+  });
+  const [ fileTree, setFileTree ] = useState<DataNode[]>([]); // Data structure to be used by the Tree
+  const [ treeMap ] = useState(() => new Map<string, string>()); // Map structure from the API
+  const [ isFetching, setIsFetching ] = useState(false);
+  const [ fileInfo, setFileInfo ] = useState<FileInfo>();
+  const [ viewMode, setViewMode ] = useState<'tree' | 'editor' | undefined>(
+    () => documentWidth <= 1024 ? 'tree' : undefined,
+  ); // To be used in the mobile view, switches the UI
+
+  const treeClasses = classNames({
+    [ css.hideElement ]:
+    (documentWidth <= 1024) && (viewMode === 'editor'),
+  });
+  const editorClasses = classNames({
+    [ css.hideElement ]:
+    (documentWidth <= 1024) && (viewMode === 'tree'),
+  });
+
+  // map the file tree
+  useEffect(() => {
+    try {
+      (async () => {
+        const files = await getExperimentFileTree({ experimentId });
+
+        const navigateTree = (node: FileNode, key: string): DataNode => {
+          treeMap.set(key, node.path || '');
+
+          const newNode: DataNode = {
+            className: 'treeNode',
+            isLeaf: true,
+            key,
+            title: node.name,
+          };
+
+          if (node.files?.length) {
+            newNode.children = node.files.map((chNode, idx) => navigateTree(chNode, `${key}-${idx}`));
+            newNode.isLeaf = false;
+          }
+
+          return newNode;
+        };
+
+        if (config) {
+          setFileTree([
+            {
+              className: 'treeNode',
+              isLeaf: true,
+              key: '0-0',
+              title: 'Configuration',
+            },
+            ...files.map<DataNode>((node, idx) => navigateTree(node, `0-${idx + 1}`)),
+          ]);
+
+          setFileInfo({
+            data: yaml.dump(config),
+            name: 'Configuration',
+            path: 'Configuration',
+          });
+
+          if (documentWidth <= 1024) { // if it's in mobile view and we have a config file available, render it as default
+            setViewMode('editor');
+          }
+        } else {
+          setFileTree(files.map<DataNode>((node, idx) => navigateTree(node, `0-${idx}`)));
+        }
+      })();
+    } catch (error) {
+      throw new Error(error as string);
+    }
+  }, [ treeMap, config, documentWidth, experimentId ]);
+
+  const onSelectFile = async (
+    keys: React.Key[],
+    info: { [key:string]: unknown, node: DataNode },
+  ) => {
+    if (info.node.title === 'Configuration') {
+      setFileInfo({
+        data: yaml.dump(config),
+        name: 'Configuration',
+        path: 'Configuration',
+      });
+
+      return;
+    }
+
+    const filePath = treeMap.get(String(keys[0])) as string;
+
+    if (filePath.includes('.')) { // check if the selected node is a file
```

Comment:
this assumption won't always work, e.g. for `ipynb_checkpoints`.

looks like the `FileNode` coming from the API has a `isDir` field, maybe you can use that instead?